src/test/java/com/training/services/tests/ProductsTest.java

```diff
@@ -62,8 +68,84 @@ public void testShorexProductTypeValidate() {
             softAssert.assertThat(
                     response.getPayload().getProducts().get(i).getProductType().getProductType()
                             .equalsIgnoreCase("SHOREX"))
-                    .as("Product id " + response.getPayload().getProducts().get(i).getProductID() +" is not SHOREX")
+                    .as("Product id " + response.getPayload().getProducts().get(i).getProductID() + " is not SHOREX")
                     .isTrue();
         }
     }
+
+    /**
+     * Checking if offeringDate, day  should be less than or equal to 10 days, from sailing date 20200809
+     *
+     * @throws ParseException
+     */
+    @Test
+    public void testShorexOfferingDateValidate() throws ParseException {
+        response = new RestEngine().getResponseGet(baseURL, headerMap, queryParam)
+                .as(Response.class);
+
+        Map<String, Integer> dayMonthDaysInMonthFromLink = getDayMonthDaysInMonth(map.get("sailingIdParamValue").split("AL")[1]);
+
+        List<Products> products = response.getPayload().getProducts();
+        List<Offering> offerings;
+
+        for (int index = 0; index < products.size(); index++) {
+             offerings = products.get(index).getOffering();
+            for (int getDateOffering = 0; getDateOffering < offerings.size(); getDateOffering++) {
+
+                Map<String, Integer> dayMonthDaysInMonth = getDayMonthDaysInMonth(offerings.get(getDateOffering).getOfferingDate());
+
+                /**
+                 * If query paramter link date and offeringDate are from same month
+                 */
+                if (dayMonthDaysInMonthFromLink.get("month") == dayMonthDaysInMonth.get("month")) {
+                    if (!((dayMonthDaysInMonth.get("day") - dayMonthDaysInMonthFromLink.get("day")) <= 10)) {
+                        softAssert.fail("Product ID = " + offerings.get(getDateOffering)
+                                .getProductID() + " offering date is greater than 10 days = "
+                                + offerings.get(getDateOffering).getOfferingDate());
+                    }
```

Comment:
instead of this, can we directly apply 
`assert.((dayMonthDaysInMonth.get("day") - dayMonthDaysInMonthFromLink.get("day"))isGreaterThanOrEqualTo(10)`?